### ISSUE TITLE
Are multiple subcommands possible?

### ISSUE THREAD

**Issue by lespea** [hellflame/argparse#1]
**Describe the bug**

Wondering if it's possible to support having multiple subcommands parsed at the same time.  Might be too tricky with positional arguments but thought I'd post an issue to start a discussion.

**To Reproduce**

Sample test code showing the behavior.  This was the most concise code I could think of covering most of the use cases (didn't include positional arguments in this test).  Main thing is it would have to support duplicated flags for the different subcommands.

```go
package argTester

import (
	"github.com/hellflame/argparse"
	"testing"
)

func TestMultiCmds(t *testing.T) {
	p := argparse.NewParser("multi subcmds", "multiple sub commands work", nil)

	c1 := false
	c2 := false

	cmd1 := "cmd1"
	sc1 := p.AddCommand(cmd1, "cmd1", &argparse.ParserConfig{
		DefaultAction: func() {
			c1 = true
		},
	})
	c1f := sc1.Flag("t", "test", nil)

	cmd2 := "cmd2"
	sc2 := p.AddCommand(cmd2, "cmd2", &argparse.ParserConfig{
		DefaultAction: func() {
			c2 = true
		},
	})
	c2f := sc2.Flag("t", "test", nil)

	if err := p.Parse([]string{cmd1, "-t", cmd2, "-t"}); err != nil {
		t.Errorf("Couldn't run: %+v", err)
	}

	if !(c1 && *c1f) {
		t.Error("Cmd1 not called")
	}
	if !(c2 && *c2f) {
		t.Error("Cmd2 not called")
	}
}

```

**Expected behavior**

Tests pass.

**Desktop (please complete the following information):**

 - OS: Any
 - Version 1.2.2 (latest)

**Comment by hellflame** [comment]
thx for the feed back, and I see what you want to achieve. 

if I understand it correctly, all I have to do is to try to match subcommand when there is no positional argument matched. So technically speaking, __it is possible to do so__

But the original idea of subcommand is to help users handle different kind of tasks one by one, and we are used to focus on only one subject at a time, one subcommand or just the main entry. 

It's quite hard for users to manage multiple subcommand arguments at the same time. if the program showed multiple subcommand entries, which means these inputs will cause multiple irrelevant actions

eg: `git add . commit -m update`, which seems convenient, but actually quite confused. 

I build this package from my develop experience and use experience, I'm not sure where have you seen use cases like multiple subcommands at the same time? I would like to check out, and I'll take it into consideration.  

if there are cases programer need the users to handle different kind of inputs, I suggest arrange the arguments by group, which would be like:

```
optional arguments:
  --help, -h                        show this help message
  --completion                      show command completion script
  --version, -v                     show version of easy-mail

Email Options:
  --from FROM, -f FROM              email send from
  --to TO, -t TO                    recv address list
  --subject SUBJECT, -s SUBJECT     email subject
  --content CONTENT, -c CONTENT     email content

Authorize Options:
  --smtp SMTP                       manually set smtp address like: smtp.abc.com:465
  --password PASSWORD, -p PASSWORD  email password
```

in conclusion, argparsing is not just parsing args, it's also for better using experience, a few restrictions will help programers build better programs.

**Comment by lespea** [comment]
I'm currently using this library to build a single exe that has many smaller functionality pieces, so allowing multiple subcommands allows the user (mostly myself) to easily compose a series of sub actions without running the binary over and over again.  I can see how it might be detrimental for some use cases.  This isn't a hard requirement as I can build "higher level" sub commands that combines the individual components but it doesn't scale at all and will only cover the combinations that I think of in advance.

If you don't want to impl this I understand it would just fit my, admittedly rather niche, use case well.

**Comment by hellflame** [comment]
frankly speaking, your case is quite complex. But I just came up a way of using the package to fit your case. Check this out

```go
func main() {
  parser := argparse.NewParser("basic", "this is a basic program", nil)

  testParser := argparse.NewParser("test", "test command", nil)
  isOk := testParser.Flag("", "ok", nil)

  parser.Strings("t", "test", &argparse.Option{Action: func(args []string) error {
    if e := testParser.Parse(args); e != nil {
      return e
    }
    fmt.Println("is it ok? ", *isOk)
    return nil
  }})

  buildParser := argparse.NewParser("build", "build command", nil)
  count := buildParser.Int("c", "count", nil)

  parser.Strings("b", "build", &argparse.Option{Action: func(args []string) error {
    if e := buildParser.Parse(args); e != nil {
      return e
    }
    fmt.Println("got an int", *count)
    return nil
  }})

  if e := parser.Parse(nil); e != nil {
    fmt.Println(e.Error())
    return
  }
}


```

**Comment by lespea** [comment]
Hmm yeah that's not a great experience, I'll just keep to my current strategy of making higher level combinations... it isn't that big of a deal.  Thanks for giving this some thought :)
